### DISCUSSION TITLE
Recommendations for deprecating private formula

### DISCUSSION THREAD

**Discussion by dandavison** [Homebrew/discussions#139] [Everyday usage] [answered]
I have a project which was distributed via a private Homebrew formula for a while, before being accepted into the public Homebrew distribution. I'd like to deprecate the private formula; is there any recommended way to do that?

**Comment by Rylan12** [comment, marked as answer]
You can use the `deprecate!` and `disable!` methods that we use in Homebrew/core (just do a quick search for those methods and plenty of examples will pop up).

If the formula was migrated to another tap, you can also use a `tap_migrations.json` file to indicate that the formula moved to another tap. If you do this, trying to install the formula will give the following message:

```console
$ brew install <formula>
Error: No available formula or cask with the name "<formula>".
It was migrated from <user>/<tap> to <new-user>/<new-tap>.
You can access it again by running:
  brew tap <new-user>/<new-tap>
And then you can install it by running:
  brew install <formula>
```

You can see [an example](https://github.com/Homebrew/homebrew-core/blob/master/tap_migrations.json) of a `tap_migrations.json` file in Homebrew/core.

**Reply by dandavison** [reply]
Thanks very much @Rylan12. I added this line

```ruby
disable! because: "it is now in homebrew core. Please reinstall it as follows:\nbrew untap dandavison/delta\nbrew install git-delta\n"
```
